apps/rush-lib/src/data/test/EnvironmentConfiguration.test.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+// <reference types='mocha' />
+
+import { assert } from 'chai';
+import {
+  EnvironmentConfiguration,
+  EnvironmentValue
+} from '../EnvironmentConfiguration';
+
+describe('EnvironmentConfiguration', () => {
+  let _oldEnv: typeof process.env;
+
+  beforeEach(() => {
+    _oldEnv = process.env;
+    process.env = {};
+  });
+
+  afterEach(() => {
+    process.env = _oldEnv;
+  });
+
+  describe('initialize', () => {
+    it('correctly allows no environment variables', () => {
+      assert.doesNotThrow(EnvironmentConfiguration.initialize);
+    });
+
+    it('allows known environment variables', () => {
+      process.env['rush_tempDir'] = '/var/temp'; // tslint:disable-line:no-string-literal
+      assert.doesNotThrow(EnvironmentConfiguration.initialize);
+    });
+
+    it('does not allow unknown environment variables', () => {
```

Comment:
>'does not allow unknown environment variab [](start = 7, length = 42)

I hope this is scoped to environment variables starting with `rush_`. Also I'm not sure this check is necessary.